tools/generators/epic-generator/index.ts

```diff
@@ -0,0 +1,191 @@
+
+  try {
+    authStatus = execSync('ghz auth status', { stdio: [] }).toString();
+  } catch (e) {
+    throw new Error(`Error calling GitHub CLI (gh). Please make sure it's installed correctly.\n${e.message}`);
+  }
+
+  if (authStatus.includes('You are not logged into any GitHub hosts')) {
+    throw new Error('You are not logged into GitHub CLI (gh).');
+  }
+};
+
+interface Package {
+  name: string;
+  folder: string;
+  owners: string[];
+}
+
+interface Ownership {
+  path: string;
+  owners: string[];
+}
+
+interface MigrationIssue {
+  issueUrl?: string;
+  assignee?: string;
+  packages: Package[];
+}
+
+type MigrationIssues = Record<string, MigrationIssue>;
+
+const getConvergedPackages = (tree: Tree) => {
+  const projects = getProjects(tree);
+
+  const convergedPackages: Package[] = [];
+  projects.forEach((project, key) => {
+    if (project.projectType === 'library' && isPackageConverged(tree, project)) {
+      convergedPackages.push({
+        name: key,
+        folder: project.root,
+        owners: [],
+      });
+    }
+  });
+
+  return convergedPackages;
+};
+
+const getCodeowners = (): Ownership[] => {
+  const codeownersContent = readFileSync('.github/CODEOWNERS', 'utf8');
+
+  return codeownersContent.split('\n').map(line => {
+    const [path, ...owners] = line.split(' ');
+
+    return {
+      path: path.replace(/\/+$/, ''),
+      owners,
+    };
+  });
+};
+
+const mapOwnerships = (packages: Package[], ownerships: Ownership[]): Package[] => {
+  return packages.map(pckg => {
+    const owners = ownerships.find(ownership => ownership.path === pckg.folder)?.owners || [];
+
+    return {
+      ...pckg,
+      owners,
+    };
+  });
+};
+
+const getPackages = (tree: Tree) => {
+  const packages = getConvergedPackages(tree);
+
+  const ownerships = getCodeowners();
+
+  return mapOwnerships(packages, ownerships);
+};
+
+const createEpic = (repo: string, title: string, message: string) => {
+  const issueUrl = execSync(`gh issue create --repo "${repo}" --title "${title}" --body "${message}"`)
+    .toString()
+    .trim();
+
+  return issueUrl;
+};
+
+const createIssue = (repo: string, issue: MigrationIssue, templateTitle: string) => {
+  const title = `${templateTitle} - ${issue.assignee || 'ownerless'}`;
+  const message = stripIndents`
+    🚧 This is an auto-generated issue to individually track migration progress.
+
+    ### Packages to migrate:
+    ${issue.packages.map(pckg => `- ${pckg.name}`).join('\n')}
+  `;
+
+  const command = `gh issue create --repo "${repo}" --title "${title}" --body "${message}"`;
+
+  const issueUrl = execSync(command).toString().trim();
+
+  return issueUrl;
+};
+
+const generateIssues = (repo: string, templateTitle: string, packages: Package[]) => {
+  const migrationIssues = packages.reduce<MigrationIssues>((acc, pckg) => {
+    const teamOwner = pckg.owners.find(owner => owner.startsWith('@microsoft/'));
+    const key = teamOwner || 'ownerless';
+
+    if (key in acc) {
```

Comment:
- `in` is dangerous operator. in this case it will be evaluated as truthy even if value is undefined.

I'd suggest to use exit early approach as branching tends to create less readable code

```
if (acc[key]) {
      acc[key].packages.push(pckg);
      return acc;
    }

    acc[key] = {
      assignee: teamOwner,
      packages: [pckg],
    };
    
    return acc;
```